ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/AuthServiceJWTPayload.scala

```diff
@@ -198,32 +228,50 @@ object AuthServiceJWTCodec {
   // ------------------------------------------------------------------------------------------------------------------
   // Decoding
   // ------------------------------------------------------------------------------------------------------------------
-  def readFromString(value: String): Try[AuthServiceJWTPayload] = {
-    for {
-      json <- Try(value.parseJson)
-      parsed <- Try(readPayload(json))
-    } yield parsed
+  def readAudienceBasedToken(value: JsValue): AuthServiceJWTPayload = value match {
+    case JsObject(fields) =>
+      StandardJWTPayload(
+        issuer = readOptionalString(propIss, fields),
+        participantId = None,
+        userId = readString(propSub, fields),
+        exp = readInstant(propExp, fields),
+        format = StandardJWTTokenFormat.Audience,
+        audiences = readOptionalStringOrArray(propAud, fields),
+        scope = readOptionalString(propScope, fields),
+      )
+    case _ =>
+      deserializationError(
+        s"Could not read ${value.prettyPrint} as AuthServiceJWTPayload: value is not an object"
+      )
   }
 
-  def readAudienceBasedToken(value: JsValue): AuthServiceJWTPayload = value match {
+  def readScopeBasedToken(value: JsValue): AuthServiceJWTPayload = value match {
     case JsObject(fields) =>
       StandardJWTPayload(
         issuer = readOptionalString(propIss, fields),
         participantId = None,
         userId = readString(propSub, fields),
         exp = readInstant(propExp, fields),
-        format = StandardJWTTokenFormat.ParticipantId,
+        format = StandardJWTTokenFormat.Scope,
         audiences = readOptionalStringOrArray(propAud, fields),
+        scope = readOptionalString(propScope, fields),
       )
     case _ =>
       deserializationError(
         s"Could not read ${value.prettyPrint} as AuthServiceJWTPayload: value is not an object"
       )
   }
 
-  def readPayload(value: JsValue): AuthServiceJWTPayload = value match {
+  def readFromString(value: String): Try[AuthServiceJWTPayload] =
+    for {
+      json <- Try(value.parseJson)
+      parsed <- Try(readPayload(json))
+    } yield parsed
+
+  private def readPayload(value: JsValue): AuthServiceJWTPayload = value match {
```

Comment:
To fit in with the `readXxxToken` naming, rename to `readToken` or `readStandardToken` ?

Reply:
I think it makes sense, but since this code is copy-pasted at the moment (and here we are looking at a copy). I would wait with the changes for the re-unification, which will happen when canton drop becomes the upstream of the daml repo